src/libappimage/type2.c

```diff
@@ -182,7 +186,7 @@ bool type2_read_file_into_buf(struct appimage_handler* handler, void* traverse,
     }
 
     // resolve symlink if possible
-    if (!appimage_type2_resolve_symlink(fs, &inode)) {
+    if (!appimage_type2_resolve_symlink(fs, &inode, 4)) {
```

Comment:
If you intend to use `4` everywhere, you should define a global constant somewhere.